### PR TITLE
Optimize DECIMAL constants for byte code generation

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/TestExpressionOptimizer.java [hunk 1]
```diff
@@ -57,11 +59,26 @@
 
 public class TestExpressionOptimizer
 {
+    private TypeRegistry typeManager;
+    private ExpressionOptimizer optimizer;
+
+    @BeforeClass
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/TestExpressionOptimizer.java]: if there nothing that need to be released, why `@BeforeClass` is needed?

File: presto-main/src/test/java/com/facebook/presto/sql/TestSqlToRowExpressionTranslator.java [hunk 2]
```diff
@@ -14,38 +14,105 @@
...
+import static com.facebook.presto.sql.planner.LiteralInterpreter.toExpression;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
 
 public class TestSqlToRowExpressionTranslator
 {
+    private Metadata metadata;
+
+    @BeforeClass
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/TestSqlToRowExpressionTranslator.java]: same comment as in previous commit

File: presto-main/src/test/java/com/facebook/presto/sql/TestSqlToRowExpressionTranslator.java [hunk 3]
```diff
@@ -14,38 +14,105 @@
...
+    {
+        return translateAndOptimize(expression, getExpressionTypes(expression));
+    }
+
+    private RowExpression translateAndOptimize(Expression expression, Map<NodeRef<Expression>, Type> types)
+    {
+        return SqlToRowExpressionTranslator.translate(expression, SCALAR, types, metadata.getFunctionRegistry(), metadata.getTypeManager(), TEST_SESSION, true);
+    }
+
+    private Expression simplifyExpression(Expression expression)
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/TestSqlToRowExpressionTranslator.java]: it is not yet used

File: presto-main/src/test/java/com/facebook/presto/sql/TestExpressionOptimizer.java (hunk 1 above)
Comment (findepi) [reply presto-main/src/test/java/com/facebook/presto/sql/TestExpressionOptimizer.java]: that is?

File: presto-main/src/test/java/com/facebook/presto/sql/TestSqlToRowExpressionTranslator.java (hunk 3 above)
Comment (findepi) [reply presto-main/src/test/java/com/facebook/presto/sql/TestSqlToRowExpressionTranslator.java]: i am aware. This makes the second commit easier to read. Do you think it's bad?

**Comment by findepi** [comment]
squashed

**Comment by losipiuk** [comment]
You may squash the last two too.

**Comment by findepi** [comment]
> You may squash the last two too.
> Edit: Naaa. Better not. Adding declarations is actually a bug fix.

Exactly. But comment edits don't go over mail 😉

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**